tools/mpremote/mpremote/repl.py

```diff
@@ -35,8 +43,15 @@ def do_repl_main_loop(
             n = state.transport.serial.inWaiting()
         except OSError as er:
             if er.args[0] == 5:  # IO error, device disappeared
-                print("device disconnected")
+                disconnected = True
+                break
+        except Exception as er:
+            # Handle Windows serial exception
+            if isinstance(er, OSError) and "ClearCommError failed" in str(er):
```

Comment:
Instead of checking if it's an instance of OSError, just move the str test up into the above `except OSError as er` block.

Reply:
Oh whoops, I just cleaned up / consolidated the exception checking but it's in thte follow-on reconnect commit not this one. I'll pull it in here.